store/sqlstore/post_store.go

```diff
@@ -1745,11 +1744,9 @@ func (s *SqlPostStore) SearchPostsInTeamForUser(paramsList []*model.SearchParams
 	pchan := make(chan store.StoreResult, len(paramsList))
 
 	for _, params := range paramsList {
-		// Don't allow users to search for everything.
-		if params.Terms == "*" {
-			continue
-		}
-
+		// remove any term that contains only non-alphanumeric chars
+		// ex: "abcd ** ef*g &^!" >> "abcd ef*g"
```

Comment:
@thefactremains Our objective here is to remove non-alphanumeric strings of characters for the post search to avoid problems in the DB search. Do you see any problem or reduction of functionality doing this?

Reply:
@ethervoid I want to make sure I understand. Search strings that contain ONLY non-alphanumeric characters will be removed?

Reply:
Yes. For example having a search string “test 123 **^%” should end as “test 123”

Reply:
I cannot think of a use-case where a user needs to search by non-alphanumeric strings, but that doesn't mean one doesn't exist. I would prefer to leave the capability to search for non-alphanumeric strings (if the quotes are required to do so, that would be acceptable).

Reply:
Ok. @amarjeetanandsingh then if the non-alphanumeric terms are enclosed with quotes, we should avoid the removal. Thank you @thefactremains

Reply:
> enclosed with quotes

@ethervoid It can be single quote `'` or double quote `"` both?

Reply:
Double quote is what we're using to treat a term as literal

Reply:
Ok.
Thanks for clarification. Will make the changes tonight.

Reply:
Great. Thank you!

Reply:
@ethervoid 
I have completed the [changes](https://github.com/amarjeetanandsingh/mattermost-server/commit/81ac7f614c48dbb7254966cd298181e89a92c631).